### PR TITLE
Platform prefix

### PR DESCRIPTION
**Pull request by zolotov**

### CODE REVIEW

**Review by hsz — APPROVED** [review]

File: src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt [hunk 1]
```diff
@@ -145,18 +124,22 @@ abstract class RunIdeBase(runAlways: Boolean) : JavaExec() {
...
+                    .asSequence()
+                    .filter { file -> file.fileName.toString().endsWith(".sh") }
+                    .flatMap { file -> Files.lines(file).asSequence() }
+                    .mapNotNull { line -> platformPrefixSystemPropertyRegex.find(line)?.groupValues?.getOrNull(1) }
+                    .firstOrNull()
+
+            if (prefix != null) {
+                systemProperty("idea.platform.prefix", prefix)
             }
+            info(context, "Using idea.platform.prefix=$prefix")
```
Comment (shalupov) [inline src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt:138]: IMHO you need to assert the case where prefix was not found by both system property and script search

Comment (zolotov) [reply src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt:138]: > where prefix was not found by both system property and script search

Unfortunately, it means that empty prefix should be used for IntelliJ Ultimate

Comment (shalupov) [reply src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt:138]: Yes, in that case, you may check that it's really IntelliJ Ultimate and not some other IDE

Comment (zolotov) [reply src/main/kotlin/org/jetbrains/intellij/tasks/RunIdeBase.kt:138]: makes sense. done

**Pull request merged**